File: MovieWebsite-backend/src/main/java/com/project/MovieWebsite/models/Ads.java
package com.project.MovieWebsite.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Table(name="tab_ads")
@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder

public class Ads {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name="name", nullable= false, length=255)
    private String name;

    @Column(name="description", nullable= false, length=255)
    private String description;

    @Column(name="banner_ads", nullable= false)
    private String bannerAds;

    @Column(name="create_at", nullable= false)
    private Date createAt;

    @Column(name="expiration_at", nullable= false)
    private Date expirationAt;

    @Column(name="amount", nullable= false)
    private int amount;

    @Column(name="is_active")
    private int isActive;

    @PrePersist
    protected void onCreate() {
        this.createAt = new Date();
    }
}

File: Website_Phim/WebPhim-Frontend/MovieWebsite-fontend/src/app/models/ads.ts
import { SafeUrl } from "@angular/platform-browser";

export interface Ads {
    id: number,
    name: string,
    description: string,
    trading_code: string,
    email: string,
    list_img: string[],
    list_img_url: SafeUrl[],
    create_at: Date,
    expiration_at: Date,
    amount: number,
    is_active: number,
    is_confirm: number,
    number_days: number
}
